zmon-controller-app/src/main/java/org/zalando/zmon/controller/MultiKairosDBController.java

```diff
@@ -189,10 +189,13 @@ private void alignQueriesToFullMinutes(final JsonNode node, final int queryWindo
                 start = start - (start % 60000);
                 q.put("start_absolute", start);
             } else if (q.has("start_relative")) {
-                if (queryWindow != 0) {
-                    ObjectNode r = (ObjectNode) q.get("start_relative");
-                    r.put("value", queryWindow);
-                    r.put("unit", "minutes");
+                if (maxQueryWindow != 0) {
+                    int curWindow = q.get("start_relative").asInt();
+                    if curWindow > maxQueryWindow {
```

Comment:
For Java, looks like `if (...)` is missing parens.

Reply:
too much go :)